addons/interaction/functions/fnc_passMagazine.sqf

```diff
@@ -12,21 +12,24 @@
  * None
  *
  * Example:
- * [_player, _target, "arifle_MX_F"] call ace_interaction_fnc_magToPassazine
+ * [_player, _target, "arifle_MX_F"] call ace_interaction_fnc_passMagazine
  *
  * Public: No
  */
 params ["_player", "_target", "_weapon"];
 
-private _compatibleMags = getArray (configfile >> "CfgWeapons" >> _weapon >> "magazines");
-private _filteredMags = magazinesAmmoFull _player select {(_x select 0) in _compatibleMags && {!(_x select 2)}};
+private _compatibleMags = [_weapon] call CBA_fnc_compatibleMagazines;
+private _filteredMags = magazinesAmmoFull _player select {
+    _x params ["_className", "", "_loaded"];
+    (_className in _compatibleMags) && {!_loaded} && {_target canAdd _className}
```

Comment:
Order by speed? !_loaded is faster to check than `in`

Reply:
`!_loaded` will be true most of the time, so I don't think we should bother checking it first